app/domain/operations/transmittable/generate_response_objects.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Operations
+  module Transmittable
+    # find or create job, then create response transmission/transaction, save payload and link to subject
+    class GenerateResponseObjects
+      include ::Operations::Transmittable::TransmittableUtils
+
+      def call(params)
+        values = yield validate_params(params)
+        @job = yield find_or_create_job_by_job_id(values)
+        transmission_params = values.merge({ job: @job, event: 'acked', state_key: :acked })
+        @transmission = yield create_response_transmission(transmission_params, { job: @job })
+        subject = yield find_subject(params[:subject_type], params[:correlation_id])
+        transaction_params = values.merge({ transmission: @transmission, subject: subject, event: 'acked', state_key: :acked })
+        @transaction = yield create_response_transaction(transaction_params, { job: @job, transmission: @transmission })
+        _transaction = yield save_payload(params[:payload])
+
+        transmittable
+      end
+
+      private
+
+      def validate_params(params)
+        return Failure('Cannot save a failure response without a payload') if params[:payload].blank?
+        return Failure('Cannot save a failure response without a key') unless params[:key].is_a?(Symbol)
+        return Failure('Cannot link a subject without a correlation_id') unless params[:correlation_id].is_a?(String)
+        return Failure('Cannot link a subject without a subject_type') unless params[:subject_type].is_a?(String)
+
+        Success({ key: params[:key],
+                  title: params[:key].to_s.humanize.titleize,
+                  correlation_id: params[:correlation_id],
+                  started_at: DateTime.now,
+                  publish_on: DateTime.now,
+                  job_id: params[:job_id]})
+      end
+
+      def find_subject(type, correlation_id)
```

Comment:
we should pass the gid of the subject, then just do `subject = GlobalID::Locator.locate(gid)` then we don't need to lay out all the conditionals